geolocator_platform_interface/lib/src/implementations/method_channel_geolocator.dart

```diff
@@ -178,6 +184,31 @@ class MethodChannelGeolocator extends GeolocatorPlatform {
     return _positionStream;
   }
 
+  @override
+  Stream<NmeaMessage> getNmeaMessageStream() {
+    if (_nmeaMessageStream != null) {
+      return _nmeaMessageStream;
+    }
+
+    var nmeaStream = nmeaChannel.receiveBroadcastStream();
```

Comment:
The `nmeaStream` variable can be made `final`, once it is initialized it does not change (and is should not change), by making it a `final` variable we can protect against this.
```suggestion
    final nmeaStream = nmeaChannel.receiveBroadcastStream();
```